src/support/generation.c

```diff
@@ -80,8 +78,7 @@ uint64_t
  *	Wait for the resource to drain.
  */
 void
-	TABBED IN to avoid dist/ functions:
-	__wt_gen_drain(WT_SESSION_IMPL *session, int which, uint64_t generation)
+__wt_gen_drain(WT_SESSION_IMPL *session, int which, uint64_t generation)
 {
```

Comment:
Can we write `__wt_gen_drain` as

```c
while (__wt_gen_oldest(session, which) < v)
    __wt_yield(); /* falling back to __wt_sleep */
```

?

It would be nice to have just one place that has to figure out when all sessions have left a generation, and assuming we cache the oldest generation, this could be more efficient than always walking.

When we talked about using read/write locks to track threads in generations, this is part of what I meant about the oldest generation catching up to current when all threads leave their generations.